### ISSUE TITLE
Cannot run with sound null safety.

### ISSUE THREAD

**Issue by inc16sec** [zino-app/graphql-flutter#858]
After migrating to null-safety and updating all packages I wanted to run the App and this error showed up.

```
Error output from Xcode build:
↳
    ** BUILD FAILED **
Xcode's output:
↳
    Error: Cannot run with sound null safety, because the following dependencies
    don't support null safety:
- package:gql
    For solutions, see https://dart.dev/go/unsound-null-safety
    ../../../../documents/flutter/.pub-cache/hosted/pub.dartlang.org/graphql_flutter-5.0.0-nullsafety.1/lib/src/widgets/query.dart:62:44: Warning: Operand of null-aware operation '!' has type 'DefaultPolicies' which excludes null.
     - 'DefaultPolicies' is from 'package:graphql/src/core/policies.dart' ('../../../../documents/flutter/.pub-cache/hosted/pub.dartlang.org/graphql-5.0.0-nullsafety.2/lib/src/core/policies.dart').
        optionsWithOverrides.policies = client.defaultPolicies!.watchQuery
```
I've tried to run the app with no sound null safety using `flutter run --no-sound-null-safety` yet I still run to some issues because the project now relies on null-safety.
Is this an issue on my part or there are packages that still need to be updated.
Thank you.

**Comment by micimize** [comment]
Are you using `graphql_flutter: ^5.0.0-nullsafety.2` or `graphqlr: ^5.0.0-nullsafety.3`?

**Comment by inc16sec** [comment]
I'm using `graphql_flutter: ^5.0.0-nullsafety.1`. Is this issue fixed on `^5.0.0-nullsafety.3`

**Comment by inc16sec** [comment]
> I'm using `graphql_flutter: ^5.0.0-nullsafety.1`. Is this issue fixed on `graphql_flutter: ^5.0.0-nullsafety.2`

I have just updated to `graphql_flutter: ^5.0.0-nullsafety.2` and I'm still getting the same error as above.

**Comment by inc16sec** [comment]
I fixed it. 
I was actually overriding `graphql_flutter: 5.0.0-nullsafety.1` and `gql:  ^0.12.4` so I can use the package Artemis. And it wa  `gql` that was causing tha app no to run as for `graphql_flutter` it was showing me warning and they went away as soon as I changed `graphql_flutter: 5.0.0-nullsafety.1` to `graphql_flutter: 5.0.0-nullsafety.2` in dependency_overrides.
like this.

```yaml
dependency_overrides:
 graphql_flutter: 5.0.0-nullsafety.1
```
to
```yaml
dependency_overrides:
 graphql_flutter: 5.0.0-nullsafety.2
```
and of course I had to also change `gql` from `gql:  ^0.12.4` to a null-safe version in my case it was
```yaml
dependency_overrides:
gql: ^0.13.0-nullsafety.2 
 graphql_flutter: 5.0.0-nullsafety.2
```
And that's it now the app is running just fine.
